src/game_battler.cpp

```diff
@@ -252,27 +286,38 @@ bool Game_Battler::UseItem(int item_id) {
 		return true;
 	}
 
-	switch (item->type) {
-		case RPG::Item::Type_weapon:
-		case RPG::Item::Type_shield:
-		case RPG::Item::Type_armor:
-		case RPG::Item::Type_helmet:
-		case RPG::Item::Type_accessory:
-			return item->use_skill && UseSkill(item->skill_id);
-		case RPG::Item::Type_special:
-			return UseSkill(item->skill_id);
+	bool do_skill = RPG::Item::Type_special
+		|| (item->use_skill && (
+				item->type == RPG::Item::Type_weapon
+				|| item->type == RPG::Item::Type_shield
+				|| item->type == RPG::Item::Type_armor
+				|| item->type == RPG::Item::Type_helmet
+				|| item->type == RPG::Item::Type_accessory
+				)
+				);
+
+	if (do_skill) {
+		auto* skill = ReaderUtil::GetElement(Data::skills, item->skill_id);
+		if (skill != nullptr) {
```

Comment:
This silent rejection of the skill can hide a game bug.

Needs a Output::Warning "IsSkillUsable: Can't use item %d with invalid skill %d"

Reply:
This has been added